### PR TITLE
165 feature local trajectory planning with frenet trajectory planner

### LINKED ISSUE #165: [Feature]: Local trajectory planning with Frenet Trajectory planner
### Description

Integrate the trajectory planner into local planning. Tune trajectory planner so it creates a valid trajectory to overtake 

### Definition of Done

- [ ] Planner generates valid trajectory (Road width, side, ...)

### PR DESCRIPTION
**Pull request by samuelkuehnel**
# Description

This PR introduces the frenet trajectory planner to our project. It enables to create a new trajecotry during runtime when a object ist detected in front. 
Fixes #165 

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Does this PR introduce a breaking change?

No
## Most important changes

New method in motion planning. 
`utils.py` file that has some utility methods to free up space in the motion planning file.
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works (might be obsolete with CI later on)
- [x] New and existing unit tests pass locally with my changes (might be obsolete with CI later on)

### CODE REVIEW

File: code/agent/config/rviz_config.rviz [hunk 1]
```diff
@@ -63,11 +63,11 @@ Visualization Manager:
       Unreliable: false
       Value: true
       Visibility:
-        Grid: false
+        Grid: true
         Imu: false
-        Path: false
+        Path: true
```
Comment (JuliusMiller) [inline code/agent/config/rviz_config.rviz:68]: Can you set this to false for the main-branch?

File: code/planning/src/global_planner/global_planner.py [hunk 2]
```diff
@@ -74,7 +74,7 @@ def __init__(self):
 
         self.path_pub = self.new_publisher(
             msg_type=Path,
-            topic='/paf/' + self.role_name + '/trajectory',
+            topic='/paf/' + self.role_name + '/trajectory_dummy',
```
Comment (JuliusMiller) [inline code/planning/src/global_planner/global_planner.py:77]: Maybe call it trajectory_global ?

File: code/planning/src/local_planner/ACC.py [hunk 3]
```diff
@@ -47,7 +47,7 @@ def __init__(self):
         # Get trajectory to determine current speed limit
         self.trajectory_sub: Subscriber = self.new_subscription(
             Path,
-            f"/paf/{self.role_name}/trajectory",
+            f"/paf/{self.role_name}/trajectory_dummy",
```
Comment (JuliusMiller) [inline code/planning/src/local_planner/ACC.py:50]: Maybe call it trajectory_global ?

File: code/planning/src/local_planner/motion_planning.py [hunk 4]
```diff
@@ -33,15 +35,54 @@ class MotionPlanning(CompatibleNode):
...
+            self.__set_heading,
+            qos_profile=1)
+        self.test_sub = self.new_subscription(
+            Bool,
+            f"/paf/{self.role_name}/test",
+            self.change_trajectory,
+            qos_profile=1)
+        self.trajectory_sub = self.new_subscription(
+            Path,
+            f"/paf/{self.role_name}/trajectory_dummy",
```
Comment (JuliusMiller) [inline code/planning/src/local_planner/motion_planning.py:78]: -> trajectory_global ?

File: code/planning/src/local_planner/motion_planning.py [hunk 5]
```diff
@@ -33,15 +35,54 @@ class MotionPlanning(CompatibleNode):
...
+            self.__get_current_velocity,
+            qos_profile=1)
+        self.head_sub = self.new_subscription(
+            Float32,
+            f"/paf/{self.role_name}/current_heading",
+            self.__set_heading,
+            qos_profile=1)
+        self.test_sub = self.new_subscription(
+            Bool,
+            f"/paf/{self.role_name}/test",
```
Comment (JuliusMiller) [inline code/planning/src/local_planner/motion_planning.py:73]: test ?

**Review by JuliusMiller — CHANGES_REQUESTED** [review]
Maybe change some things for better understanding on the main-branch

**Comment by samuelkuehnel** [comment]
Changes done please re-review

**Review by JuliusMiller — APPROVED** [review]
Nice Changes! Approved!

**Pull request merged**